src/plugins/vis_type_table/public/agg_table/agg_table.js

```diff
@@ -71,15 +76,22 @@ export function KbnAggTable(config, RecursionHelper) {
           return val;
         }
 
-        // escape each cell in each row
-        const csvRows = rows.map(function (row) {
-          return Object.entries(row).map(([k, v]) => {
+        let csvRows = [];
+        rows.map((row) => {
+          const rowArray = [];
+          Object.entries(row).map(([k, v]) => {
             const column = columns.find((c) => c.id === k);
-            if (formatted && column) {
-              return escape(column.formatter.convert(v));
-            }
-            return escape(v);
+
+            if (!column) return;
```

Comment:
Have a question to understand this part. The change in this implementation is, if there's no column available for a rows's property, don't add it to the column. This sounds valid, and an improvement. Just try to understand, how could that be the case? Are there row properties with no columns? Just checking to prevent some kind of regression here. Because then there is a nice potential  to simplify the code here in a way like this

```Typescript
for (const col of columns) {
   const value = row[col.id];
   const formattedValue = formatted && col.formatter
      ? escape(col.formatter.convert(v))
      : escape(v);
   rowArray.push(formattedValue);
}

```

Reply:
Yes there is a case where the user has selected two buckets and has ticked the Show partial rows. On the table you can't see any change but when the CSV is downloaded an extra column is added but as there is no extra header so it is messed up. So the question here is which is the expected behavior? As far as I can't see any change on the table, what should I see on the CSV?

Reply:
synced with @stratoula, and she showed me another issue in the current implementation, fixed by this PR, so here the example described in the recent comment, 

<img width="1950" alt="Screenshot 2020-05-26 at 5 52 01 PM" src="https://user-images.githubusercontent.com/463851/82920619-c992ff00-9f77-11ea-9a42-cb10c5d69642.png">
Exporting this table led to an formatted CSV export with an extra column

```
"extension.keyword: Descending","geo.dest: Descending",Count
,624,122,CN
,624,96,IN
,624,51,US
,624,23,ID
,624,22,PK
gz,285,63,CN
gz,285,52,IN
gz,285,16,US
gz,285,9,NG
gz,285,7,ID
```
This is also fixed by this PR, and therefore it's safe to refactor the existing code in the suggested way